Sources/NIOSOCKS/Messages/ClientGreeting.swift

```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+/// The SOCKS handshake begins with the client sending a greeting
+/// containing supported authentication methods.
+public struct ClientGreeting: Hashable {
```

Comment:
Is there any reason for this to be `public`?

Reply:
Yes - the server implementer will need to receive the greeting to have a look at the authentication methods.

Reply:
Ok, great. Can I propose that we start by making everything `internal` that can be for this proposal, and then widen the access in the next PR? I appreciate it adds some noise but it reduces the risk that we miss something in API review that accidentally makes it into the public API.

Reply:
Sure!

Reply:
Can this go back to being `internal`?